src/pip/_internal/utils/misc.py

```diff
@@ -1337,7 +1340,14 @@ def protect_pip_from_modification_on_windows(modifying_pip):
         new_command = [
             sys.executable, "-m", "pip"
         ] + sys.argv[1:]
-        raise CommandError(
+        error_message = (
             'To modify pip, please run the following command:\n{}'
             .format(" ".join(new_command))
         )
+        if as_dependency:
+            error_message = (
+                'pip is being modified because it is '
+                'listed as a dependency. ' +
```

Comment:
I reckon it's possible to do something along the lines of:

```
                'listed as a dependency of {parent_requirements}. {message}'
```

Reply:
Do you mean `parent_requirements=req_pip.comes_from`?

Reply:
Or even `req_pip.comes_from.name`.

Reply:
I meant more like resolver.reverse_dependencies['pip'] which, of course, isn't implemented (yet!). 🙃

Comes from, IIRC, is a string we compose to show the sequence that lead us to this package? I used to know the nuances of this, but I've since paged out this knowledge. I really think that this needs a better name. Would you be willing to look into that as a follow up to this PR?

Coming back, I'm a little weary of adding it to the message directly because it's not complete information. I guess it'll be okay because the main intent here is probably to tell the user that "run again and fix technical issues"?

The only case I can imagine is if someone drops the printed dependency for some reason, it'd cause confusion when there might be something else depending on pip, which is a pretttty edge-casey edge case.

Reply:
Sorry for not answering so long.

> Comes from, IIRC, is a string we compose to show the sequence that lead us to this package? I used to know the nuances of this, but I've since paged out this knowledge. I really think that this needs a better name. Would you be willing to look into that as a follow up to this PR?

Would you like me to open an issue? I'd love to.

> Coming back, I'm a little weary of adding it to the message directly because it's not complete information. I guess it'll be okay because the main intent here is probably to tell the user that "run again and fix technical issues"?
> 
> The only case I can imagine is if someone drops the printed dependency for some reason, it'd cause confusion when there might be something else depending on pip, which is a pretttty edge-casey edge case.

Honestly, If I faced the issue, I wouldn't care about the dependency (which causes the problem). I want to continue somehow, and it's nice that pip suggests a solution for that. The current message is okay as for me. It could be rephrased though if you like.

What do you suggest? Should we close the PR and leave the message as is, or try to add all dependencies?

Reply:
@pradyunsg kindly ping 👋